resources/views/project/subproject-dependencies.blade.php

```diff
@@ -1,118 +1,11 @@
-            if (a.name < b.name) {
-                return -1;
-            }
-            if (a.name > b.name) {
-                return 1;
-            }
-            return 0;
-        }
-
-        function sort_by_id (a, b) {
-            if (a.id < b.id) {
-                return -1;
-            }
-            if (a.id > b.id) {
-                return 1;
-            }
-            return 0;
-        }
-
-        function mouseOvered(d) {
-            var header1Text = "Name: " + d.key;
-            if (d.group !== undefined) {
-                header1Text += ", Group: " + d.group;
-            }
-            $('#header1').html(header1Text);
-            if (d.depends) {
-                var depends = "<p>Depends: ";
-                depends += d.depends.join(", ") + "</p>";
-                $('#dependency').html(depends);
-            }
-            var dependents = "";
-            d3.selectAll('.node--source').each(function (p) {
-                if (p.key) {
-                    dependents += p.key + ", ";
-                }
-            });
-
-            if (dependents) {
-                dependents = "Dependents: " + dependents.substring(0,dependents.length-2);
-                $('#dependents').html(dependents);
-            }
-            d3.select("#toolTip").style("left", (d3.event.pageX + 40) + "px")
-                .style("top", (d3.event.pageY + 5) + "px")
-                .style("opacity", ".9");
-        }
-
-        function mouseOuted(d) {
-            $('#header1').text("");
-            $('#dependents').text("");
-            $('#dependency').text("");
-            d3.select("#toolTip").style("opacity", "0");
-        }
-
-        function resetDepView() {
-            d3.select('#chart_placeholder svg').remove();
-            d3.select('#chart_placeholder')
-                .datum(dataroot)
-                .call(chart);
-        }
-        $(function(){
-            $('#selectedsort').on("change", function(e) {
-                selected = $(this).val();
-                if (parseInt(selected) === 1) {
-                    dataroot.sort(sort_by_id);
-                } else if (parseInt(selected) === 0) {
-                    dataroot.sort(sort_by_name);
-                }
-                resetDepView(dataroot);
-            });
-            var ajaxlink = "{{ url('ajax/getsubprojectdependencies.php') }}?project={{ urlencode($project->Name) }}";
-            d3.json(ajaxlink, function(error, classes) {
-                if (error){
-                    errormsg = "json error " + error + " data: " + classes;
-                    console.log(errormsg);
-                    document.write(errormsg);
-                    return;
-                }
-                dataroot = classes;
-                dataroot.sort(sort_by_name);
-                resetDepView(dataroot);
-            });
-        });
-    </script>
+    <sub-project-dependencies
+        :project-name="'{{ urlencode($project->Name) }}'"
```

Comment:
I believe the urlencode is not necessary here.  You can encode it whenever you need it, rather than passing around something for which it might not be clear whether it has been escaped or not.